packages/core/src/Machine.ts

```diff
@@ -142,11 +139,5 @@ export function createMachine<
   TServiceMap,
   TTypesMeta
 > {
-  if (!IS_PRODUCTION && !config.predictableActionArguments && !warned) {
```

Comment:
I think a better change here would be something along those lines:
```diff
-if (!IS_PRODUCTION && !config.predictableActionArguments && !warned) {
+if (!IS_PRODUCTION && !(predictableActionArguments in config) && !warned) {
```

Reply:
If it prevents the excess warnings, feel free to make the change on this branch (or a new one); I'm open for any improvement

Reply:
This change wouldn't prevent the excess warnings - it would allow people to silence the warning by using `predictableActionArguments: false`. 

We could also add a test-env detection (like [here](https://github.com/emotion-js/emotion/blob/92be52d894c7d81d013285e9dfe90820e6b178f8/packages/react/src/index.js#L20)) and skip the warnings in those envs. This should help with excessive warnings to some extent.

Reply:
@Andarist I made these changes